components/TextFields/src/MDCTextInputControllerBase.m

```diff
@@ -1398,24 +1398,31 @@ - (void)updateLayout {
  underlineLabelsOffset                                                // Depends on text insets mode
  */
 // clang-format on
-- (UIEdgeInsets)textInsets:(UIEdgeInsets)defaultInsets {
+- (UIEdgeInsets)textInsets:(UIEdgeInsets)defaultInsets
+    withSizeThatFitsWidthHint:(CGFloat)widthHint {
   // NOTE: UITextFields have a centerY based layout. And you can change EITHER the height or the Y.
   // Not both. Don't know why. So, we have to leave the text rect as big as the bounds and move it
   // to a Y that works. In other words, no bottom inset will make a difference here for UITextFields
   UIEdgeInsets textInsets = defaultInsets;
 
-  if (!self.isFloatingEnabled) {
-    return textInsets;
+  if (self.isFloatingEnabled) {
+    textInsets.top = MDCTextInputControllerBaseDefaultPadding +
+                     MDCRint(self.textInput.placeholderLabel.font.lineHeight *
```

Comment:
Should this be MDCCeil? I'd expect us to always need to create a ceiling of space for the line.

Reply:
Agreed! We spoke offline and I'll be leaving this as is, as this was moved from below and changing this can cause visual breaking changes.